PhysicsTools/NanoAOD/interface/SimpleFlatTableProducer.h

```diff
@@ -163,7 +163,7 @@ class SimpleFlatTableProducer : public SimpleFlatTableProducerBase<T, edm::View<
     } else {
       for (unsigned int i = 0, n = prod->size(); i < n; ++i) {
         const auto &obj = (*prod)[i];
-        if (cut_(obj)) {
+        if (prod->ptrAt(i).isNonnull() && prod->ptrAt(i).isAvailable() && cut_(obj)) {
```

Comment:
I would remove the isAvailable, otherwise this would cause NanoAOD to fail silently if it's misconfigured to read from a collection that has been dropped (e.g. if one skims some collection by reference, then drops the original, and tries to make a table from the skim)

Reply:
I tried that, but it doesn't catch the dangling pointers because they aren't null (they are garbage).

Reply:
Pointers shouldn't be garbage in CMSSW, they should either be null or invalid because they point to a dropped collection => I'm assuming it's the latter case.
I don't understand why this fix is related to that:  in #30112 the problem is daughters of a GenJet (and it's a feature, not a bug), here prod is a collection read from the event, so one level less deep.
If in your workflow you're first taking the daughters of the jet and putting them in a collection and then running the flat table on top of it, then IMO you should put the fix in the piece of code that is dong the first task, not here.

Reply:
OK, that's also fine. We can adjust the `UniqueMerger` class instead. We can close this PR and start a new one with the updates.